programs/local/LocalServer.cpp

```diff
@@ -387,14 +397,12 @@ int LocalServer::mainImpl()
         /// Use the same query_id (and thread group) for all queries
         CurrentThread::QueryScope query_scope_holder(query_context);
 
-        if (need_render_progress && !is_interactive)
+        if (need_render_progress)
         {
             /// Set progress callback, which can be run from multiple threads.
             query_context->setProgressCallback([&](const Progress & value)
             {
-                /// Write progress only if progress was updated
-                if (progress_indication.updateProgress(value))
-                    progress_indication.writeProgress();
+                onProgress(value);
```

Comment:
This is too frequent, it does not looks really nice right now (too twitchy)

Reply:
It seems to me as twitchy as it is on master in clickhouse-client for `select * from numbers(1e12)`, isn't it? And there is zero twitchyness in clickhouse-local progress for query with no output like `select count() from numbers(1e12)`. 

The twitchyness is caused by this line https://github.com/ClickHouse/ClickHouse/blob/6e8eb598bfb007ae10827c923391d8b7dfa25d81/programs/local/LocalServer.cpp#L240
and removing it removes absolutely all flickering. But if removed, there will be mess between progress ouput and query output.

Reply:
I guess you meant this line - https://github.com/ClickHouse/ClickHouse/pull/26231/files#diff-e606245fd4a9bed218932b9548398a29fb2834f7d4bdd683b241b40d01516829R284, i.e. from the `flush_buffer_func` callback.

And it seems that my query is pretty exotic, so I think that flickering in this case is fine.